inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/SubjectObjectFeatureEditor.java

```diff
@@ -293,43 +294,61 @@ protected IJQueryTemplate newTemplate()
 
     private void setSelectedKBItem(KBHandle value)
     {
-        if (roleLabelIsFilled()) {
-            setFeatureValueInCas(value);
+        // We do not want to store the error handle.
+        if (KBHandle.isErrorHandle(value)) {
+            return;
         }
-    }
-
-    private void setFeatureValueInCas(KBHandle value) {
-        try {
-            JCas jCas = actionHandler.getEditorCas();
-            AnnotationFS selectedFS = WebAnnoCasUtil.selectByAddr(jCas, roleModel.targetAddr);
-            WebAnnoCasUtil.setFeature(selectedFS, linkedAnnotationFeature,
-                value != null ? value.getIdentifier() : value);
-        }
-        catch (Exception e) {
-            LOG.error("Error: " + e.getMessage(), e);
-            error("Error: " + e.getMessage());
+        
+        if (roleLabelIsFilled()) {
+            try {
+                JCas jCas = actionHandler.getEditorCas();
+                AnnotationFS selectedFS = WebAnnoCasUtil.selectByAddr(jCas, roleModel.targetAddr);
+                WebAnnoCasUtil.setFeature(selectedFS, linkedAnnotationFeature,
+                    value != null ? value.getIdentifier() : value);
+            }
+            catch (Exception e) {
+                error("Error: " + e.getMessage());
+                LOG.error("Error: " + e.getMessage(), e);
+            }
         }
     }
 
     private KBHandle getSelectedKBItem()
     {
         KBHandle selectedKBHandleItem = null;
         if (roleLabelIsFilled()) {
+            String selectedKBItemIdentifier = null;
+            
             try {
                 JCas jCas = actionHandler.getEditorCas();
                 AnnotationFS selectedFS = WebAnnoCasUtil.selectByAddr(jCas, roleModel.targetAddr);
-                String selectedKBItemIdentifier = WebAnnoCasUtil
+                selectedKBItemIdentifier = WebAnnoCasUtil
                     .getFeature(selectedFS, linkedAnnotationFeature.getName());
-
+            }
+            catch (Exception e) {
+                LOG.error("Error: " + e.getMessage(), e);
+                // We cannot use feedback messages in code that is called from the load() method
+                // of a LoadableDetachableModel, so this is an alternative way of passing the
+                // error on to the user.
+                selectedKBHandleItem = KBHandle.errorHandle("Unable to load CAS", e);
```

Comment:
Why no early return here?

Reply:
Sometimes I tend to avoid returns in the middle of a method because it's like a goto.

Reply:
From what I see in the error case, you will check whether `selectedKBHandleItem ``is not null, which it is not in the error case, then you will try to `getKBInstancesByIdentifierAndTraits`, which should error again. I would just error once.

Reply:
I might be missing something, but the second case shouldn't be triggered since the selectedKBItemIdentifier will remain null. 

I'm rearranging the code in https://github.com/inception-project/inception/pull/268 - please have another look there. Thanks!

Reply:
You are right, these are two selected thingies, sorry for bothering you!

Reply:
No worries. Better double check than having bad code... as with the extra throws I forgot to remove ;)